system/jlib/jsuperhash.cpp

```diff
@@ -483,26 +483,36 @@ void SuperHashTable::doKill(void)
 
 void SuperHashTable::releaseAll(void)
 {
-    unsigned i;
-    for (i = 0; i < tablesize; i++)
+    if (tablecount)
     {
-        void * et = table[i];
-        table[i] = NULL;
-        if (et)
-            onRemove(et);
+        unsigned i;
+        for (i = 0; i < tablesize; i++)
+        {
+            void * et = table[i];
+            table[i] = NULL;
+            if (et)
+                onRemove(et);
+        }
+        tablecount = 0;
+        setCache(0);
     }
-    tablecount = 0;
+}
+
+void SuperHashTable::releaseElements()
+{
+    releaseAll();
 }
 
 void SuperHashTable::kill(void)
 {
     releaseAll();
-    doKill();
-    tablesize = InitialTableSize;
-    tablecount = 0;
-    setCache(0);
-    table = (void * *) checked_malloc(InitialTableSize*sizeof(void *),-604);
-    memset(table,0,InitialTableSize*sizeof(void *));
+    if (tablesize != InitialTableSize)
```

Comment:
avoid kill() reallocating a table that is already the correct size (e.g., if it has already been killed)